tensorflow-examples/src/main/java/org/tensorflow/model/examples/objectdetection/FasterRcnnInception.java

```diff
@@ -225,78 +258,71 @@ public static void main(String[] params) {
 
             DecodeJpeg.Options options = DecodeJpeg.channels(3L);
             DecodeJpeg decodeImage = tf.image.decodeJpeg(readFile.contents(), options);
+
             //fetch image from file
             TUint8 outputImage = (TUint8) runner.fetch(decodeImage).run().get(0);
             Shape imageShape = outputImage.shape();
+            //dimensions of test image
             long[] shapeArray = imageShape.asArray();
-
+            //reshape the tensor to 4D for input to model
+            Reshape<TUint8> reshape = tf.reshape(tf.constant(outputImage),
+                    tf.array(1,
+                            outputImage.shape().asArray()[0],
+                            outputImage.shape().asArray()[1],
+                            outputImage.shape().asArray()[2]
+                    )
+            );
+            TUint8 reshapeTensor = (TUint8) runner.fetch(reshape).run().get(1);
+            Map<String, Tensor> feedDict = new HashMap<>();
             //The given SavedModel SignatureDef input
-            Map<String, Tensor> feed_dict = new HashMap<>();
-            feed_dict.put("input_tensor", reshapeTensor(outputImage));
+            feedDict.put("input_tensor", reshapeTensor);//reshapeTensor(outputImage));
             //The given SavedModel MetaGraphDef key
-            Map<String, Tensor> outputTensorMap = model.function("serving_default").call(feed_dict);
+            Map<String, Tensor> outputTensorMap = model.function("serving_default").call(feedDict);
```

Comment:
I think this is going to leak the remaining tensors, as there are more outputs that you aren't closing. At the moment the output of a function isn't autocloseable, so you'll need to iterate the map and close each element.

Reply:
Will do